components/forms/conditionOfJoining/CojView.tsx

```diff
@@ -25,21 +29,128 @@ export default function CojView() {
   const {
     signingCojProgName: progName,
     signingCojSignedDate: signedDate,
-    signingCoj
+    signingCoj,
+    signingCojVersion
   } = store.getState().user;
 
   if (!signingCoj) return <Redirect to="/programmes" />;
+
   return progName ? (
     <>
       {signedDate && <FormSavePDF history={history} path={"/programmes"} />}
       <ScrollTo />
-      <CojVersion progName={progName} />
-      <CojDeclarationSection signedDate={signedDate} />
+      <CojVersion progName={progName} version={signingCojVersion} />
+      {signingCojVersion === "GG9" && (
+        <CojDeclarationSection9 signedDate={signedDate} />
+      )}
+      {signingCojVersion === "GG10" && (
+        <CojDeclarationSection10 signedDate={signedDate} />
+      )}
     </>
   ) : null;
 }
 
-function CojDeclarationSection({
+function CojDeclarationSection9({
```

Comment:
Hi @DorisWongNHS , to reduce the code duplication I would focus on making a more generic CojDeclarationSection component rather than 2 separate CojDeclarationSection9/10 components. You can just pass in props for any differences e.g. initialValues, validationSchema (where the only difference is `isDeclareContacted` ?), COJ_DECLARATIONS...

Reply:
Thank you JohnO, I will have a try!

Reply:
Great stuff! A possible implementation for the yup validation schema object prop is in the `formAValidationSchema.ts`  (scroll to line 101) where you can have a default schema object then have any additional properties e.g. isDeclareContacted as the rest.